nanopub/profile.py

```diff
@@ -20,46 +14,86 @@ class ProfileError(RuntimeError):
     pass
 
 
-def get_orcid_id():
-    orcid_id = get_profile()[ORCID_ID]
-    if not orcid_id:
-        raise ProfileError('Your profile was not setup yet or not setup correctly. To setup '
-                           'your profile see instructions in Readme.md')
-    return orcid_id
+class Profile:
+    """Represents a user profile.
+
+    Attributes:
+        orcid_id: The user's ORCID
+        name: The user's name
+        public_key: Path to the user's public key
+        private_key: Path to the user's private key
+        nanopub_uri: URI of the user's profile nanopub
+    """
+    def __init__(
+            self,
+            orcid_id: str, name: str,
+            public_key: Path, private_key: Path,
+            nanopub_uri: Optional[str] = None
+            ) -> None:
+        """Create a Profile."""
+        self.orcid_id = orcid_id
+        self.name = name
+        self.public_key = public_key
+        self.private_key = private_key
+        self.nanopub_uri = nanopub_uri
+
+    @classmethod
+    def _yatiml_savorize(cls, node: yatiml.Node) -> None:
+        node.rename_attribute('profile_nanopub', 'nanopub_uri')
+
+    @classmethod
+    def _yatiml_sweeten(cls, node: yatiml.Node) -> None:
+        node.remove_attributes_with_default_values(cls)
+        node.rename_attribute('nanopub_uri', 'profile_nanopub')
+
+
+_load_profile = yatiml.load_function(Profile)
+
 
+_dump_profile = yatiml.dump_function(Profile)
 
-def get_public_key():
-    filepath = get_profile()[PUBLIC_KEY_FILEPATH]
-    with open(filepath, 'r') as f:
-        public_key = f.read()
-    if not public_key:
-        raise ProfileError('Your profile was not setup yet or not setup correctly. To setup '
-                           'your profile see instructions in Readme.md')
-    return public_key
+
+def get_orcid_id() -> str:
+    """Returns the user's ORCID."""
+    return get_profile().orcid_id
+
+
+def get_public_key() -> str:
+    """Returns the user's public key."""
+    return get_profile().public_key
 
 
 @lru_cache()
-def get_profile() -> Dict[str, any]:
-    """
-    Retrieve nanopub user profile. By default the profile is stored in `HOME_DIR/.nanopub/profile.yaml`.
+def get_profile() -> Profile:
+    """Retrieve nanopub user profile.
 
-    Returns: A dict with all settings stored in the user profile yaml file.
+    By default the profile is stored in `HOME_DIR/.nanopub/profile.yaml`.
 
+    Returns:
+        A Profile containing the data from the configuration file.
+
+    Raises:
+        yatiml.RecognitionError: If there is an error in the file.
     """
-    path = PROFILE_PATH
-    with path.open('r') as f:
-        return yaml.load(f)
+    try:
+        return _load_profile(PROFILE_PATH)
+    except yatiml.RecognitionError as e:
+        raise ProfileError(str(e))
```

Comment:
I think we should tell users that their profile was not setup correctly and point them to the documentation.

Reply:
What would the documentation suggest to do? Regenerate the profile? That would overwrite the user's keys and leave them e.g. unable to retract their previous nanopubs. So presumably some kind of manual fix would be needed, in which case it would be useful to know where the problem is exactly.

Reply:
I've added some text to that effect, but also left the detailed error message so users have the information needed to fix things.

Reply:
Yeah, regenerate the profile. Overwriting the user key is optional. The solution you have now is perfect I think (both detailed and user friendly)!